File: src/a6s/lang/typescript/testdata/query_constraints.ts
// Fixture for constrained query tests (c5t note 7d1244e0, Task 1).
// Every "should NOT" case is a regression guard for the bare-query bugs.

// === Top-level declarations (SHOULD be extracted) ===
export class TopClass {
  method(): void {}
  field: number = 0;
}

class InternalClass {
  helper(): string { return "x"; }
}

export interface TopInterface {
  ifaceMethod(x: string): boolean;
}

export type TopType = string;
type InternalType = { debug: boolean };

export enum TopEnum {
  A = "a",
  B = "b",
}

export function topFunction(): void {}
function internalFunction(): void {}

export const TOP_CONST = 42;
const INTERNAL_CONST = "secret";
export let TOP_LET = 0;

// === Object-literal methods (should NOT be extracted) ===
// The headline bug: bare method_definition query matched these.
const obj = {
  objMethod(): void {},
  objMethod2() { return 1; },
};

const handler = {
  handle(req: string): void {},
};

// === Local variables (should NOT be extracted) ===
// The is_top_level band-aid fixed these; now the query handles it.
function withLocals() {
  const localConst = 1;
  let localLet = 2;
  var localVar = 3;
}

// === Nested declarations (should NOT be extracted as top-level) ===
function withNested() {
  function nestedFn(): void {}
  class NestedClass {}
  interface NestedIface { x: number; }
  enum NestedEnum { X, Y }
  type NestedType = string;
}

// === Nested class with methods (should NOT be extracted) ===
function withNestedClass() {
  class NestedClassWithMethods {
    nestedMethod(): void {}
    nestedField: number = 0;
  }
}

// === Inline object type method signatures (should NOT be extracted) ===
// is_inside_inline_object_type band-aid fixed these; now the query handles it.
export type WithInline = {
  inlineMethod(): void;
  cleanup?(): void;
};

export type NestedInline = {
  inner: {
    process(): void;
  };
};

// === Class-body method signature overloads (SHOULD be extracted as interface_method) ===
class WithOverload {
  method(x: string): string;
  method(x: number): number;
  method(x: any): any { return x; }
}

File: src/a6s/lang/kotlin/testdata/query_constraints.kt
// Fixture for constrained query tests (c5t note 7d1244e0, Task 2).
// Every "should NOT" case is a regression guard for the bare-query bugs.

package com.example.constraints

// === Top-level declarations (SHOULD be extracted) ===
class TopClass {
    fun memberFun(): String = "x"
    val memberVal: Int = 0
    var memberVar: String = ""
}

object TopObject {
    fun objectFun(): Unit {}
    const val CONST_IN_OBJECT: Int = 1
}

interface TopInterface {
    fun ifaceMethod(x: String): Boolean
}

class WithCompanion {
    companion object {
        fun companionFun(): Unit {}
        const val COMPANION_CONST: Int = 2
    }
}

enum class TopEnum {
    RED,
    GREEN,
    BLUE
}

typealias TopTypeAlias = Map<String, Int>

fun topLevelFun(): Unit {}
fun String.topLevelExtensionFun(): Int = this.length

val topLevelVal: String = "top"
var topLevelVar: Int = 0
const val TOP_LEVEL_CONST: Int = 42

// === Local functions and properties (should NOT be extracted) ===
// The Kotlin equivalent of the TS local-variable bug. Bare queries matched
// these because function_declaration / property_declaration appear inside
// function bodies too.
fun withLocals() {
    fun localFun(): Unit {}
    val localVal = 1
    var localVar = 2
    fun String.localExtension(): Int = 0
}

// === Nested classes (SHOULD be extracted — existing test contract) ===
class Outer {
    class Nested {
        fun nestedMethod(): String = "nested"
    }
    inner class Inner {
        fun innerMethod(): String = "inner"
    }
}
